src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp

```diff
@@ -288,10 +300,14 @@ void DistributedSearchTreeImpl<DeviceType>::deviseStrategy(
   // on forwarding queries to leafless trees.
   using Access = Traits::Access<Predicates, Traits::PredicatesTag>;
   auto const n_queries = Access::size(queries);
-  Kokkos::View<int *, DeviceType> new_offset(offset.label(), n_queries + 1);
+  Kokkos::View<int *, DeviceType> new_offset(
+      Kokkos::view_alloc(offset.label(), Kokkos::WithoutInitializing),
+      n_queries + 1);
   Kokkos::parallel_for(
       ARBORX_MARK_REGION("bottom_trees_with_required_cumulated_leaves_count"),
-      Kokkos::RangePolicy<ExecutionSpace>(0, n_queries), KOKKOS_LAMBDA(int i) {
+      Kokkos::RangePolicy<ExecutionSpace>(space, 0, n_queries),
+      KOKKOS_LAMBDA(int i) {
+        new_offset(i) = 0;
```

Comment:
Please make any code optimizations a separate PR>

Reply:
actually this one I might have messed up